src/apprt/gtk/Window.zig

```diff
@@ -1032,15 +1036,25 @@ fn gtkActionReset(
 }
 
 fn gtkHideWidget(_: ?*c.GtkWidget, data: ?*anyopaque) void {
-    const widget: *c.GtkWidget = @ptrCast(@alignCast(data));
-    c.gtk_widget_set_visible(widget, c.FALSE);
-    App.crash_reports_notice_dismissed = true;
+    const self = userdataSelf(data.?);
+    for (self.app.core_app.surfaces.items) |surface| {
+        const gtk_surface: *Surface = @ptrCast(surface);
+        if (gtk_surface.container.window()) |window| {
+            c.gtk_widget_set_visible(window.crash_notice_widget, c.FALSE);
+        }
+    }
+    self.app.crash_reports_notice_dismissed = true;
 }
 
 fn adwHideBanner(_: ?*c.GtkWidget, data: ?*anyopaque) void {
-    const banner: *c.AdwBanner = @ptrCast(data);
-    c.adw_banner_set_revealed(banner, c.FALSE);
-    App.crash_reports_notice_dismissed = true;
+    const self = userdataSelf(data.?);
+    for (self.app.core_app.surfaces.items) |surface| {
+        const gtk_surface: *Surface = @ptrCast(surface);
+        if (gtk_surface.container.window()) |window| {
+            c.adw_banner_set_revealed(@ptrCast(window.crash_notice_widget), c.FALSE);
```

Comment:
Even though it probably doesn't matter to the GTK code, `window.crash_notice_widget` should be checked for `null`.